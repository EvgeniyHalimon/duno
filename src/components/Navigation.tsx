import React,{useEffect} from "react";

import { Link } from 'react-router-dom';

import { Box } from "@mui/material";
import { useTypesSelector } from '../hooks/useTypesSelector';

export const Navigation: React.FC = () => {
    const {isAnime} = useTypesSelector(state => state.anime)

    useEffect(() => {
    
    },[isAnime])

    return(
        <Box sx={{ flexGrow: 1 }}>
            <ul>
                <li><Link to='/'>Home</Link></li>
                <li><Link to={isAnime ? '/popular+anime' : '/popular+manga'}>Popular</Link></li>
                <li><Link to='/genres'>Genres</Link></li>
            </ul>
        </Box>
    )
}