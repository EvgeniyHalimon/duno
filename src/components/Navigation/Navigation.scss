.navigation{
  list-style-type: none;
  padding-left: 10px;

  &-link{
    text-decoration: none;
    font-size: 20px;

    &:visited{
      color: inherit;
    }

    &:hover{
      color: #2496FF;
    }
  }
}

.sidebar{
  padding-right: 20px;
  border-right: 1px solid #2496FF;
}

.active{
  color: #2496FF !important;
  text-decoration: line-through;
}