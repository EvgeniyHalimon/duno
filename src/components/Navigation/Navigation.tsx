import React from "react";

import { Link } from 'react-router-dom';

import { Box } from "@mui/material";

import './Navigation.scss' 

export const Navigation: React.FC<any> = () => {

    return(
        <Box sx={{ flexGrow: 1 }} className='sidebar'>
            <ul className="navigation">
                <li>
                    <Link 
                        to='/'
                        className="navigation-link"
                    >
                        HOME</Link>
                </li>
                <li>
                    <Link 
                        to={'/popular'}
                        className="navigation-link"
                    >
                        POPULAR</Link>
                </li>
                <li>
                    <Link 
                        to='/genres'
                        className="navigation-link"
                    >
                        GENRES</Link>
                </li>
            </ul>
        </Box>
    )
}