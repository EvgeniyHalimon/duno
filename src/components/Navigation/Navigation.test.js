import {Link, Route, Routes, useLocation} from 'react-router-dom'
import {render, screen, fireEvent} from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import '@testing-library/jest-dom'
import {BrowserRouter, MemoryRouter} from 'react-router-dom'
import { Navigation } from './Navigation'

export const LocationDisplay = () => {
  const location = useLocation()

  return <div data-testid="location-display">{location.pathname}</div>
}

export const App = () => (
  <div>
    <Navigation/>

    <LocationDisplay />
  </div>
)



test('full app rendering/navigating', async () => {
  render(<App />, {wrapper: BrowserRouter})

  fireEvent.click(screen.getByText("POPULAR"))
  expect(screen.getByTestId('location-display')).toHaveTextContent('/popular');
})

test('Create snap', () => {
    const govna = render(<BrowserRouter><App /></BrowserRouter>)
    expect(govna.asFragment()).toMatchSnapshot()
})
