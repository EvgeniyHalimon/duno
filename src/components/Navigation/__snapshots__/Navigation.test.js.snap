// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create snap 1`] = `
<DocumentFragment>
  <div>
    <div
      class="sidebar MuiBox-root css-i9gxme"
    >
      <ul
        class="navigation"
      >
        <li>
          <a
            class="navigation-link"
            href="/"
          >
            HOME
          </a>
        </li>
        <li>
          <a
            aria-current="page"
            class="navigation-link active"
            href="/popular"
          >
            POPULAR
          </a>
        </li>
        <li>
          <a
            class="navigation-link"
            href="/genres"
          >
            GENRES
          </a>
        </li>
      </ul>
    </div>
    <div
      data-testid="location-display"
    >
      /popular
    </div>
  </div>
</DocumentFragment>
`;
